lib/Math/Matrix.pm6

```diff
@@ -51,6 +52,15 @@ method zero(Math::Matrix:U: Positive_Int $rows, Positive_Int $cols = $rows) {
     self.bless( rows => self!zero_array($rows, $cols), row-count => $rows, column-count => $cols );
 }
 
+method submatrix(Math::Matrix:U: Int $row, Int $col --> Math::Matrix:D ){
+    fail "$row is not an existing row index" unless 0 <= $row < $!row-count;
+    fail "$col is not an existing column index" unless 0 <= $col < $!column-count;
+    my @clone = @!rows.clone;
+    @clone.splice($row,1);
+    @clone = map { $^r.splice($col, 1); $^r }, @clone;
+    Math::Matrix.new( @clone );
+}
```

Comment:
not sure to understand that one, let take matrix a, like following:

1|2|3|4|5
---|---|---|---|---
6|7|8|9|10
11|12|13|14|15
16|17|18|19|20

a.submatrix( 1,2) should be a without row 1 and column 2? like that?:

1|2|4|5
---|---|---|---|---
11|12|14|15
16|17|19|20

Am i understanding right?
it would correspond to that definition: https://proofwiki.org/wiki/Definition:Submatrix

However, i was planning to create a method submatrix( begin_row, begin_column, end_row, end_column ), 
with the same starting matrix a, we would have, a.submatrix(1,1,3,5) which would keep everything between cell 1,1 and 3,5, that is:

7|8|9|10
---|---|---|---
12|13|14|15
17|18|19|20
 
That definition is also a possible one apparently, from for instance there: http://mathworld.wolfram.com/Submatrix.html

Need some clarification there, which one should we implement? both? with what name? as the name is referred for both definition apparently